extract.py

```diff
@@ -109,11 +93,63 @@ def main(args):
         sys.exit(1)
 
     title = chrome.title.replace("/", "_").replace("\\", "_")
-    chrome.quit()
-
+    title = f"{folder}{title}"
     if args.download:
-        subprocess.call(["yt-dlp", "-o", title + ".%(ext)s", mpd_list[-1]])
+        subprocess.call(["yt-dlp", "--no-overwrites", "-o", title + ".%(ext)s", mpd_list[-1]])
+
+
+def main(args):
+    """Extracts the MPD URL from the given URL using Selenium and optionally downloads the video with yt-dlp"""
+    url = (
+        args.url
+        if args.url != ""
+        else input("Enter URL (e.g.: https://www.ceskatelevize.cz/porady/10640154216-most/214381477720001/): ")
+    )
+
+    options = Options()
+
+    options.add_argument("window-size=1920x1080")
+    if not args.no_headless:
+        options.add_argument("--headless")
+    options.add_argument("--ignore-certificate-errors")
+    options.add_argument("--disable-gpu")
+    options.add_argument("--mute-audio")
+    options.add_argument("--log-level=3")
+    options.add_argument("--disable-web-security")  # allows to bypass CORS
+    options.add_argument("--disable-site-isolation-trials")
+
+    options.add_experimental_option("excludeSwitches", ["enable-logging"])
+
+    chrome = webdriver.Chrome(options=options)
+
+    #check if the URL is link to epizode list or only one epizode
+    if re.match(r"https://www.ceskatelevize.cz/[^/]*/[^/]*/$", url):
+        chrome.get(url)
+        print(f"Get infro from episode list {url}")
+
+        while True:
+            try:
+                more_button = chrome.find_element(By.CLASS_NAME, 'moreButton-0-2-141')
+                WebDriverWait(chrome, 10).until(EC.element_to_be_clickable((By.CLASS_NAME, 'moreButton-0-2-141')))
+                more_button.click()
+                WebDriverWait(chrome, 5)
+            except (NoSuchElementException, StaleElementReferenceException):
+                break
+
+        episodeListSection = chrome.find_element(By.ID, 'episodeListSection')
+        elements = episodeListSection.find_elements(By.CLASS_NAME, "ctco_1gy3thf4")
+        links = [element.get_attribute('href') for element in elements]
+        s_title = chrome.title.replace("/", "_").replace("\\", "_")
+        folder = f"./{s_title}/"
+        if not os.path.exists(folder):
+            os.makedirs(folder)
```

Comment:
```suggestion
        folder_title = chrome.title.replace("/", "_").replace("\\", "_").replace(" ", "_")
        matches = re.findall(r'[\w]', folder_title)
        folder_title = ''.join(matches) + "/"
        os.makedirs(folder_title, exist_ok=True)
```
To vytvoreni slozky jde zkratit, coz je spis malickost, ale hlavne k tomu titlu - kdyz to jeste nematchnes regexem nebo nejak nesanitizujes, tak mi to nefunguje `OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: './Most - iVysílání | Česká televize/'`